mesonbuild/modules/gnome.py

```diff
@@ -900,8 +916,10 @@ def _get_scanner_ldflags(ldflags: T.Iterable[str]) -> T.Iterable[str]:
         KwargInfo('identifier_prefix', ContainerTypeInfo(list, str), default=[], listify=True),
         KwargInfo('include_directories', ContainerTypeInfo(list, (str, build.IncludeDirs)), default=[], listify=True),
         KwargInfo('includes', ContainerTypeInfo(list, (str, GirTarget)), default=[], listify=True),
+        KwargInfo('install_gir', (bool, NoneType), since='0.61.0'),
         KwargInfo('install_dir_gir', (str, bool, NoneType),
             validator=lambda x: 'as boolean can only be false' if x is True else None),
```

Comment:
I think there's a deprecated_values keyword argument you can use for bool here

Reply:
Nice, it works.

Sad and unfortunate, its type annotations claim it only supports `T.Dict[str, str]` which is a bit bogus as by definition `KwargInfo.deprecated_values` should accept the same keys as `KwargInfo.types`.

Using this suggestion reports the following warning:

```
meson.build:584: WARNING: Project targeting '>= 0.61.0' but tried to use feature deprecated since '0.61.0': "gnome.generate_gir" keyword argument "install_dir_gir" value "False".

WARNING: Deprecated features used:
 * 0.61.0: {'"gnome.generate_gir" keyword argument "install_dir_gir" value "False"'}
```

Using the custom FeatureNew I wrote locally...

```
meson.build:584: WARNING: Project targeting '>= 0.61.0' but tried to use feature deprecated since '0.61.0': install_dir_gir false to disable installation. Use install_gir instead

WARNING: Deprecated features used:
 * 0.61.0: {'install_dir_gir false to disable installation'}
```

What I like about that is the extra_message telling people what they *should* do.

Reply:
Will probably rebase on #9687

Reply:
Rebased, looks very pretty now. Thanks!